tests/specs/stn/with-replay-error-mode.e2e.js

```diff
@@ -0,0 +1,152 @@
+ * Right now, Trace can only be in error mode when its stn flag is 0 but replay runs in error mode.
+ */
+import { testRumRequest } from '../../../tools/testing-server/utils/expect-tests'
+import { config, MODE } from '../session-replay/helpers'
+import { notIE, onlyChrome, supportsMultipleTabs } from '../../../tools/browser-matcher/common-matchers.mjs'
+
+const getTraceMode = () => browser.execute(function () {
+  const agent = Object.values(newrelic.initializedAgents)[0]
+  return [
+    agent.runtime.session.state.sessionTraceMode,
+    agent.features.session_trace.featAggregate.isStandalone
+  ]
+})
+
+describe.withBrowsersMatching(notIE)('Trace error mode', () => {
+  let initSTReceived, getFirstSTPayload, getReplayOnErrorUrl
+  beforeEach(async () => {
+    await browser.destroyAgentSession()
+    await browser.testHandle.scheduleReply('bamServer', {
+      test: testRumRequest,
+      permanent: true,
+      body: JSON.stringify({ stn: 0, err: 1, ins: 1, spa: 1, sr: 1, loaded: 1 })
+    })
+
+    initSTReceived = undefined
+    getFirstSTPayload = browser.testHandle.expectResources().then(resPayload => initSTReceived = resPayload)
+    getReplayOnErrorUrl = browser.testHandle.assetURL('stn/instrumented.html', config({ session_replay: { sampleRate: 0, errorSampleRate: 1 }, session_trace: { harvestTimeSeconds: 2 } }))
+  })
+
+  function simulateErrorInBrowser () { // this is a way to throw error in WdIO / Selenium without killing the test itself
+    const errorElem = document.createElement('script')
+    errorElem.textContent = 'throw new Error("triggered! 0__0");'
+    document.body.append(errorElem)
+  }
+
+  it('switches to full mode when an error happens after page load', async () => {
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.waitForAgentLoad()
+    await browser.waitUntil(() => browser.execute(async function () { // here for no-retry wdio stability
+      return await Object.values(newrelic.initializedAgents)[0].features.session_trace.onAggregateImported
+    }), { timeout: 5000 })
+    let startingMode = (await getTraceMode())[0]
+    expect(initSTReceived).toBeUndefined()
+    expect(startingMode).toEqual(MODE.ERROR)
+
+    await browser.execute(simulateErrorInBrowser)
+    await getFirstSTPayload
+    let modeAfterErr = (await getTraceMode())[0]
+    expect(modeAfterErr).toEqual(MODE.FULL)
+    // The loadEventEnd is part of PT and PNT resource entry and is a node created at start of page life.
+    expect(initSTReceived.request.body.res.find(node => node.n === 'loadEventEnd')).toEqual(expect.objectContaining({ n: 'loadEventEnd', o: 'document', t: 'timing' }))
+
+    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy() // double check there's nothing wrong with full mdoe interval harvest
+  })
+
+  it('starts in full mode when an error happens before page load', async () => {
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.execute(simulateErrorInBrowser)
```

Comment:
i'm a little skeptical that this actually triggers before the load event, if thats really what you are testing.  You'd be better off just putting an error script in the head of the test asset if that is important

Reply:
But does scriptstring still exist in wdio? I thought it got removed entiretly in favor of browser.execute? I have to dig around for whatever it's called now @patrickhousley do you have that answer on hand?

Reply:
nvm, I'll just find or create a new asset file ig